oasislmf/pytools/gulmc/manager.py

```diff
@@ -0,0 +1,886 @@
+                damage_bins['interpolation'],
+                eff_vuln_cdf_Ndamage_bins,
+                damage_bins[eff_vuln_cdf_Ndamage_bins - 1]['bin_to'],
+            )
+
+            losses[MAX_LOSS_IDX, item_i] = max_loss
+            losses[CHANCE_OF_LOSS_IDX, item_i] = chance_of_loss
+            losses[TIV_IDX, item_i] = exposureValue
+            losses[STD_DEV_IDX, item_i] = std_dev
+            losses[MEAN_IDX, item_i] = gul_mean
+
+            # compute random losses
+            if sample_size > 0:
+                if do_correlation:
+                    # use correlation definitions to draw correlated random values
+                    item_corr_data = corr_data_by_item_id[item['item_id']]
+                    rho = item_corr_data['correlation_value']
+
+                    if rho > 0:
+                        peril_correlation_group = item_corr_data['peril_correlation_group']
+
+                        get_corr_rval(
+                            eps_ij[peril_correlation_group], vuln_rndms_base[rng_index],
+                            rho, arr_min, arr_max, arr_N, norm_inv_cdf,
+                            arr_min_cdf, arr_max_cdf, arr_N_cdf, norm_cdf, sample_size, z_unif
+                        )
+                        vuln_rndms = z_unif
+
+                    else:
+                        vuln_rndms = vuln_rndms_base[rng_index]
+
+                else:
+                    # do not use correlation
+                    vuln_rndms = vuln_rndms_base[rng_index]
+
+                for sample_idx in range(1, sample_size + 1):
+                    if effective_damageability:
+                        # use the effective damageability cdf as the vulnerability cdf
+                        vuln_prob_to = eff_vuln_cdf[eff_vuln_cdf_i:eff_vuln_cdf_i + eff_vuln_cdf_Ndamage_bins]
+                        Ndamage_bins = eff_vuln_cdf_Ndamage_bins
+
+                    else:
+                        # use the full monte carlo approach: draw samples from the hazard intensity distribution first
+
+                        if Nbins == 1:
+                            # if hazard intensity has no uncertainty, there is no need to sample
+                            haz_bin_idx = 0
+
+                        else:
+                            # if hazard intensity has a probability distribution, sample it
+
+                            # cap `haz_rval` to the maximum `haz_prob_to` value (which should be 1.)
+                            haz_rval = haz_rndms[rng_index][sample_idx - 1]
+
+                            if debug == 1:
+                                # store the random value used for the hazard intensity sampling instead of the loss
+                                losses[sample_idx, item_i] = haz_rval
+                                continue
+
+                            if haz_rval >= haz_prob_to[Nbins - 1]:
+                                haz_rval = haz_prob_to[Nbins - 1] - 0.00000003
```

Comment:
I think haz_rval is not used after you set it here

Reply:
that's right `haz_rval` is not used after here. I'll drop line 540.

Reply:
done